homeassistant/components/tuya/base.py

```diff
@@ -47,10 +47,14 @@ def step_scaled(self) -> float:
 
     def scale_value(self, value: float) -> float:
         """Scale a value."""
+        if not isinstance(value, float):
+            value = float(value)
```

Comment:
Not fully understanding what is going on here?

This seems wrong, as these methods are only called in case the data point have been described by Tuya as already numeric. So, having them at this point, feels like working around a different issue?

The linked issue contains no debug or diagnostic information, nor does this PR.
Can you provide the more information?

Please note, that `isinstance` checks are relatively expensive and should be avoided if possible.

../Frenck

Reply:
Hello Frenck,
If you look at the issue, the issue is that sometimes there are strings passed to these functions instead of a float.

I could force the float conversion regardless (and thus removing the isinstance part) to make it faster.

This is a rare case that sometimes occur with some lighting from Tuya...

If you need more info don't hesitate to tell me :)

Have a nice day,
Azerty

Reply:
I also had imagined fixing the source origin of the bug (in the light.py file) but I thought that making the base more robust would be smarter and avoid other future problems like this one...

Tell me what you think of it please :)

Have a nice day,
Azerty

Reply:
> the issue is that sometimes there are strings passed to these functions instead of a float.

I understand what the error means, the question is: Why does that happen in the first place. This PR seems to fight a different issue.

> If you need more info don't hesitate to tell me :)

I did request more info in my previous comment and also in the issue linked.

../Frenck

Reply:
You are right, this should be fixed upstream :'(.

Most likely in the tuya/light.py at around line 651.

I'll follow up on the diagnostics and once I confirm this I'll make a new PR that addresses this correctly.

Sorry for the wasted time :'(.

Have a nice day,
Azerty